### PR TITLE
[dhcp_relay] Validate DHCPv6 multicast packets can be trapped to CPU

### PR DESCRIPTION
**Pull request by w1nda**
### Description of PR

Summary:
To make sure DHCPv6 multicast packets can be received by device, we add a test to send DHCPv6 multicast packet with specific transaction ID from server to device, and check if `tcpdump` can capture the packet.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?
We need to make sure DHCPv6 multicast packets can be received by device.
#### How did you do it?
We add a test to send DHCPv6 multicast packet with specific transaction ID from server to device, and check if `tcpdump` can capture the packet.
#### How did you verify/test it?
For mx topo, verified on Nokia-7215 testbed with 202305 image and 202205 image.

### CODE REVIEW

File: docs/api_wiki/README.md [hunk 1]
```diff
@@ -147,6 +147,10 @@ def test_fun(duthosts, rand_one_dut_hostname, ptfhost):
 
 - [get_bgp_neighbors_per_asic](sonichost_methods/get_bgp_neighbors_per_asic.md) - This command provides a summary of the bgp neighbors peered with the DUT for each asic. Returns a dictionary of dictionary that maps the BGP address for each neighbor to another dictionary listing information on that neighbor device for each asic namespace.
 
+- [add_acl_table](sonichost_methods/add_acl_table.md) - Add new acl table via command `sudo config acl add table `
```
[the remark this answers is not in the document]
Comment (w1nda) [reply docs/api_wiki/README.md]: Yes, fixed

File: docs/api_wiki/sonichost_methods/add_acl_table.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Arguments
+ - `table_name` - table name of acl table
+    - Required: `True`
+    - Type: `String`
+ - `table_type` - table type of acl table
+    - Required: `True`
+    - Type: `String`
+ - `acl_stage` - acl stage
+    - Required: `False`
```
[the remark this answers is not in the document]
Comment (w1nda) [reply docs/api_wiki/sonichost_methods/add_acl_table.md:30]: Yes, added

File: docs/api_wiki/sonichost_methods/add_acl_table.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    - Type: `String`
+        - Validate value: "ingress" or "egress"
+ - `bind_ports` - ports to bind
```
[the remark this answers is not in the document]
Comment (w1nda) [reply docs/api_wiki/sonichost_methods/add_acl_table.md:34]: Yes, added

File: tests/common/devices/sonic.py [hunk 4]
```diff
@@ -1333,6 +1333,23 @@ def check_intf_link_state(self, interface_name):
         intf_status = self.show_interface(command="status", interfaces=[interface_name])["ansible_facts"]['int_status']
         return intf_status[interface_name]['oper_state'] == 'up'
 
+    def get_intf_link_local_ipv6_addr(self, intf):
+        """
+        Get the link local ipv6 address of the interface
+
+        Args:
+            intf: The interface name
```
[the remark this answers is not in the document]
Comment (w1nda) [reply tests/common/devices/sonic.py]: Yes, changed

File: tests/common/utilities.py [hunk 5]
```diff
@@ -1071,3 +1073,38 @@ def get_egress_queue_pkt_count_all_prio(duthost, port):
...
+        % (interface, pcap_save_path, pkts_filter)
+    tcpdump_pid = duthost.shell(cmd_capture_pkts)["stdout"]
+    cmd_check_if_process_running = "ps -p %s | grep %s |grep -v grep | wc -l" % (tcpdump_pid, tcpdump_pid)
+    pytest_assert(duthost.shell(cmd_check_if_process_running)["stdout"] == "1",
+                  "Failed to start tcpdump on DUT")
+    logging.info("Start to capture packet on DUT, tcpdump pid: %s, pcap save path: %s, with command: %s"
+                 % (tcpdump_pid, pcap_save_path, cmd_capture_pkts))
+    try:
+        yield
+        time.sleep(1)
```
[the remark this answers is not in the document]
Comment (w1nda) [reply tests/common/utilities.py]: Yes, added

File: tests/dhcp_relay/test_dhcp_pkt_recv.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+DHCP_RELAY_FEATRUE_NAME = "dhcp_relay"
+DHCPV6_MAC_MULTICAST = "33:33:00:01:00:02"
+DHCPV6_IP_MULTICAST = "ff02::1:2"
+DHCPV6_UDP_CLIENT_PORT = 546
+DHCPV6_UDP_SERVER_PORT = 547
+
+
+@pytest.fixture(scope="module", autouse=True)
+def skip_dhcp_pkt_recv_tests(duthost):
```
[the remark this answers is not in the document]
Comment (w1nda) [reply tests/dhcp_relay/test_dhcp_pkt_recv.py]: Yes, changed

File: tests/dhcp_relay/test_dhcp_pkt_recv.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    features_state, _ = duthost.get_feature_status()
+    if DHCP_RELAY_FEATRUE_NAME not in features_state or features_state[DHCP_RELAY_FEATRUE_NAME] != "enabled":
```
[the remark this answers is not in the document]
Comment (lizhijianrd) [reply tests/dhcp_relay/test_dhcp_pkt_recv.py:31]: `"enabled" not in features_state.get(DHCP_RELAY_FEATRUE_NAME, "")`

**Review by lizhijianrd — APPROVED** [review]

File: tests/dhcp_relay/test_dhcp_pkt_recv.py [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+import pytest
+import random
+
+from ptf import testutils
+from scapy.layers.dhcp6 import DHCP6_Solicit
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import capture_and_check_packet_on_dut
+
+pytestmark = [
+    pytest.mark.topology('t0', 'm0', 'mx')
```
Comment (Blueve) [inline tests/dhcp_relay/test_dhcp_pkt_recv.py:12]: let's mark `mx` only

**Review by Blueve — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Pull request merged**